sr-back/src/app/modules/neighbour-profile/neighbour-profile.controller.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Inject,
+  Param,
+  Post,
+  Put,
+  Query,
+  Req,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { ApiTags } from '@nestjs/swagger';
+import {
+  NeighbourProfileAlreadyExists,
+  NeighbourProfileNotFound,
+  NeighbourProfileService,
+} from './neighbour-profile.service';
+import { User } from '../user/entities/user.entity';
+import { Request } from 'express';
+import { CreateNeighbourProfileInput } from './dto/create-neighbour-profile.dto';
+import { UpdateNeighbourProfileInput } from './dto/update-neighbour-profile.dto';
+import { MapErrorToHTTP } from 'src/common/decorators/MapErrorToHTTP';
+import { AreaNotFoundError } from 'src/area/area.service';
+import {
+  SerializeTo,
+  SerializeWithPagingTo,
+} from 'src/common/decorators/SerializeTo';
+import {
+  NeighbourProfilesDTO,
+  SingleNeighbourProfileDTO,
+} from './dto/neighbour-profile.dto';
+import { PageOptionsDTO } from 'src/common/dto/pagination.dto';
+
+@Controller('neighbour-profile')
+@ApiTags('NeghbourProfile')
+export class NeighbourProfileController {
+  @Inject()
+  private readonly neighbourProfileService: NeighbourProfileService;
+
+  @Get(':id')
+  @SerializeTo(SingleNeighbourProfileDTO)
+  @MapErrorToHTTP(NeighbourProfileNotFound, UnauthorizedException)
+  async getNeighbourProfile(@Param('id') id: number) {
+    return await this.neighbourProfileService.getNeighbourProfile(id);
+  }
```

Comment:
Забыл в контроллере переделать, тут дожно быть что-то типа:

```suggestion
  async getCurrentUserProfile(@Param('id') userId: number) {
    return await this.neighbourProfileService. getNeighbourProfileByUserId(userId);
  }
```